go/processor/floria/run_context_test.go

```diff
@@ -340,17 +341,220 @@ func TestCreateAddress(t *testing.T) {
+				require.ErrorContains(t, err, test.expectedError.Error())
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestCreate_FinalizeCreateSetsCodeOrResetsResult(t *testing.T) {
+	tests := map[string]struct {
+		revision      tosca.Revision
+		code          []byte
+		gas           tosca.Gas
+		success       bool
+		resultGasLeft tosca.Gas
+	}{
+		"successful": {
+			revision:      tosca.R07_Istanbul,
+			code:          []byte{1, 2, 3},
+			gas:           tosca.Gas(601),
+			success:       true,
+			resultGasLeft: tosca.Gas(1),
+		},
+		"greater max code size": {
+			revision:      tosca.R07_Istanbul,
+			code:          make([]byte, maxCodeSize+1),
+			gas:           tosca.Gas(100000000),
+			success:       false,
+			resultGasLeft: tosca.Gas(0),
+		},
+		"starts with 0xEF pre Shanghai": {
+			revision:      tosca.R07_Istanbul,
+			code:          append([]byte{0xEF}, []byte{1, 2, 3}...),
+			gas:           tosca.Gas(801),
+			success:       true,
+			resultGasLeft: tosca.Gas(1),
+		},
+		"starts with 0xEF Shanghai": {
+			revision:      tosca.R12_Shanghai,
+			code:          append([]byte{0xEF}, []byte{1, 2, 3}...),
+			gas:           tosca.Gas(801),
+			success:       false,
+			resultGasLeft: tosca.Gas(0),
+		},
+		"too little gas": {
+			revision:      tosca.R07_Istanbul,
+			code:          []byte{1, 2, 3},
+			gas:           tosca.Gas(599),
+			success:       false,
+			resultGasLeft: tosca.Gas(0),
+		},
+	}
+
 	for name, test := range tests {
 		t.Run(name, func(t *testing.T) {
-			var want tosca.Address
-			if test.kind == tosca.Create {
-				want = tosca.Address(crypto.CreateAddress(common.Address(test.sender), test.nonce))
+			ctrl := gomock.NewController(t)
+			context := tosca.NewMockTransactionContext(ctrl)
+			snapshot := tosca.Snapshot(42)
+
+			createdAddress := tosca.Address{1}
+			result := tosca.Result{
+				Success: true,
+				Output:  test.code,
+				GasLeft: test.gas,
+			}
+
+			if test.success {
+				context.EXPECT().SetCode(createdAddress, tosca.Code(test.code))
 			} else {
-				want = tosca.Address(crypto.CreateAddress2(common.Address(test.sender), common.Hash(test.salt), test.initHash[:]))
+				context.EXPECT().RestoreSnapshot(snapshot)
+			}
+
+			finalizedResult := finalizeCreate(result, createdAddress, snapshot, test.revision, context)
+
+			result.GasLeft = test.resultGasLeft
+			if !test.success {
+				result.Output = nil
+				result.Success = false
+			}
+			require.Equal(t, result, finalizedResult, "Finalized result does not match expected result")
+		})
+	}
+}
+
+func TestCreate_PreChecksReturnsErrorOnNonceOverflowOrTransferFailure(t *testing.T) {
+	tests := map[string]struct {
+		nonce uint64
+		value tosca.Value
+		err   error
+	}{
+		"nonce overflow": {
+			nonce: math.MaxUint64,
+			err:   fmt.Errorf("nonce overflow"),
+		},
+		"insufficient balance": {
+			nonce: 0,
+			value: tosca.NewValue(1000),
+			err:   fmt.Errorf("insufficient balance"),
+		},
+		"successful": {
```

Comment:
small nit: maybe even a personal preference, try to avoid mixing successful case and error cases in the same test